src/linemod_detect.cpp

```diff
@@ -277,10 +277,16 @@ struct Detector: public object_recognition_core::db::bases::ModelReaderBase {
       }
       if (*use_depth_)
       {
+        if (depth_->depth() == 5)
+        {
+          *depth_ *= 1000.0f;
+          depth_->convertTo(*depth_, CV_16UC1);
```

Comment:
convertTo can have a scale argument: please use to have everything in only one operation